pkg/controllers/state/cluster.go

```diff
@@ -293,6 +296,37 @@ func (c *Cluster) Reset() {
 	c.nameToProviderID = map[string]string{}
 	c.bindings = map[types.NamespacedName]string{}
 	c.antiAffinityPods = sync.Map{}
+	c.daemonSetPods = map[types.NamespacedName]*v1.Pod{}
+}
+
+func (c *Cluster) GetDaemonSetPods(daemonset *appsv1.DaemonSet) *v1.Pod {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	key := types.NamespacedName{Name: daemonset.Name, Namespace: daemonset.Namespace}
+	if pod, ok := c.daemonSetPods[key]; ok {
```

Comment:
```suggestion
	if pod, ok := c.daemonSetPods[client.ObjectKeyFromObject(daemonset)]; ok {
```

We don't need the above line